### PR TITLE
Null merge from PS-5.7.27 at commit 'cf74174a7f9' into PS-8.0.17

### PR DESCRIPTION
**Pull request by inikep**
Add main.bug94176 MTR test
***
Improve RocksDB and TokuDB compilation with clang and gcc

### CODE REVIEW

**Comment by inikep** [comment]
Tested:
`--big-test` at https://ps80.cd.percona.com/job/percona-server-8.0-param/385/
`--big-test -DWITH_TOKUDB=ON -DWITH_ROCKSDB=ON` at https://ps80.cd.percona.com/job/percona-server-8.0-param/386/

File: storage/rocksdb/CMakeLists.txt [hunk 1]
```diff
@@ -27,11 +27,16 @@ IF (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
...
 ENDIF ()
 
-# Relax unused variable warnings for release modes
-MY_CHECK_CXX_COMPILER_FLAG("-Wunused-variable" HAVE_UNUSED_VARIABLE)
-IF(HAVE_UNUSED_VARIABLE)
-  SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Wno-unused-variable")
-  SET(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Wno-unused-variable")
+# Suppress warnings for gcc-9
+IF(CMAKE_COMPILER_IS_GNUCXX AND
+   (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 9.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0))
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt:32]:
```suggestion
   NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
```

File: storage/rocksdb/CMakeLists.txt [hunk 2]
```diff
@@ -27,11 +27,16 @@ IF (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
...
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-copy -Wno-redundant-move -Wno-pessimizing-move -Wno-sign-compare")
+ENDIF()
+
+# Suppress warnings for clang-8
+IF((CMAKE_CXX_COMPILER_ID MATCHES "Clang") AND
+   (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 8.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8.0))
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt:38]:
```suggestion
   NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.0)
```

File: storage/rocksdb/CMakeLists.txt [hunk 3]
```diff
@@ -44,20 +49,19 @@ IF(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 ENDIF()
 
 # Relax unused variable warnings
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-variable")
-SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable")
+MY_CHECK_CXX_COMPILER_FLAG("-Wno-unused-variable" HAVE_NO_UNUSED_VARIABLE)
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt:52]: The same append_cflags_if_supported()?

File: storage/tokudb/CMakeLists.txt [hunk 4]
```diff
@@ -78,9 +78,17 @@ IF(DEFINED TOKUDB_NOPATCH_CONFIG)
     ADD_DEFINITIONS("-DTOKUDB_NOPATCH_CONFIG=${TOKUDB_NOPATCH_CONFIG}")
 ENDIF()
 
-prepend_cflags_if_supported(-Wno-missing-field-initializers)
+append_cflags_if_supported(-Wno-missing-field-initializers)
 append_cflags_if_supported(-Wno-vla)
 
+# Disable warnings for gcc-9 or higher
+IF(CMAKE_COMPILER_IS_GNUCXX AND
+    (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 9.0 OR
```
Comment (percona-ysorokin) [inline storage/tokudb/CMakeLists.txt:86]:
```suggestion
   NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
```

**Review by percona-ysorokin — APPROVED** [review]
LGTM with minor comments

File: storage/rocksdb/CMakeLists.txt (hunk 1 above)
Comment (inikep) [reply storage/rocksdb/CMakeLists.txt:32]: Good catch, but we already merged VERSION_EQUAL OR VERSION_GREATER in 5.7

**Pull request merged**